src/main/java/org/kohsuke/github/GHWorkflowRun.java

```diff
@@ -0,0 +1,352 @@
+    }
+
+    /**
+     * The jobs URL, like https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/jobs
+     *
+     * @return the diff url
+     */
+    public URL getJobsUrl() {
+        return GitHubClient.parseURL(jobsUrl);
+    }
+
+    /**
+     * The logs URL, like https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/logs
+     *
+     * @return the diff url
+     */
+    public URL getLogsUrl() {
+        return GitHubClient.parseURL(logsUrl);
+    }
+
+    /**
+     * The check suite URL, like https://api.github.com/repos/octo-org/octo-repo/check-suites/414944374
+     *
+     * @return the diff url
+     */
+    public URL getCheckSuiteUrl() {
+        return GitHubClient.parseURL(checkSuiteUrl);
+    }
+
+    /**
+     * The artifacts URL, like https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/artifacts
+     *
+     * @return the diff url
+     */
+    public URL getArtifactsUrl() {
+        return GitHubClient.parseURL(artifactsUrl);
+    }
+
+    /**
+     * The cancel URL, like https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/cancel
+     *
+     * @return the diff url
+     */
+    public URL getCancelUrl() {
+        return GitHubClient.parseURL(cancelUrl);
+    }
+
+    /**
+     * The rerun URL, like https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/rerun
+     *
+     * @return the diff url
+     */
+    public URL getRerunUrl() {
+        return GitHubClient.parseURL(rerunUrl);
+    }
+
+    /**
+     * The workflow URL, like https://api.github.com/repos/octo-org/octo-repo/actions/workflows/159038
+     *
+     * @return the diff url
+     */
+    public URL getWorkflowUrl() {
+        return GitHubClient.parseURL(workflowUrl);
+    }
+
+    /**
+     * The head branch name the changes are on.
+     *
+     * @return head branch name
+     */
+    public String getHeadBranch() {
+        return headBranch;
+    }
+
+    /**
+     * Gets the HEAD SHA.
+     *
+     * @return sha for the HEAD commit
+     */
+    public String getHeadSha() {
+        return headSha;
+    }
+
+    /**
+     * The commit of current head.
+     *
+     * @return head commit
+     */
+    public HeadCommit getHeadCommit() {
+        return headCommit;
+    }
+
+    /**
+     * The repository of current head.
+     *
+     * @return head repository
+     */
+    public GHRepository getHeadRepository() {
+        return headRepository;
+    }
+
+    /**
+     * The type of event that triggered the build.
+     *
+     * @return type of event
+     */
+    public GHEvent getEvent() {
+        return Enum.valueOf(GHEvent.class, event.toUpperCase(Locale.ROOT));
+    }
+
+    /**
+     * Gets status of the workflow run. It can be one of queued, in_progress, or completed.
+     *
+     * @return status of the workflow run
+     */
+    // TODO GHWorkflowRun should we rather return a String here? That's what GHCheckSuite does.
```

Comment:
Here, I'm not consistent with GHCheckSuite but I'm not entirely sure it's a good idea to return a String here.

My natural inclination would be for consistency but having an enum is better from a usability POV. Or did you do that to be sure it never fails even if GitHub adds new values?

WDYT?

Reply:
What you've done here is reasonable.  Store as `String` but return an Enum.  This will result in a runtime exception in the string doesn't match the enum, but the object can still be loaded and interacted with in other ways.  

I'm pretty sure `GHCheckSuite` is the "never fails even if GitHub adds a new value".   We see this problem most strikingly in `GHEvent` where they keep adding new types and it causes hard failures.  

Consistency might also be achieved by changing `GHCheckSuite` to match this. 

In an ideal world, we'd use a "graceful enum" that stores the literal string value and returns the matching enum value or a configurable fallback value when the string doesn't match any of the enum values.  That would solve the issue we see with `GHEvent` and also what we see here.  So, for example, you'd get a `Status.UNKNOWN` if they ever add a new value that doesn't parse, and there'd need to be a way to get the raw string value at that point.

Reply:
But can we break compatibility for `GHCheckSuite`?

If not, it's probably better to have both consistent for now and open an issue for a 2.0 to change them to enum (with a fallback and a `getRaw...` to get the string value).

Reply:
> But can we break compatibility for `GHCheckSuite`?

We could add a `@BridgeMethod` to keep binary compatibility.  Are they the same status and conclusion enumerations?